src/policy/concrete.rs

```diff
@@ -137,30 +143,109 @@ impl fmt::Display for PolicyError {
 }
 
 impl<Pk: MiniscriptKey> Policy<Pk> {
-    /// Single-Node compilation
+    /// Create a Huffman Tree from compiled [Miniscript] nodes
+    #[cfg(feature = "compiler")]
+    fn with_huffman_tree<T>(
+        ms: Vec<(OrdF64, Miniscript<Pk, Tap>)>,
+        f: T,
+    ) -> Result<TapTree<Pk>, Error>
+    where
+        T: Fn(OrdF64) -> OrdF64,
+    {
+        // Pattern match terminal Or/ Terminal (with equal odds)
+        let mut node_weights = BinaryHeap::<(Reverse<OrdF64>, Arc<TapTree<Pk>>)>::new();
+        for (prob, script) in ms {
+            node_weights.push((Reverse(f(prob)), Arc::from(TapTree::Leaf(Arc::new(script)))));
+        }
+        if node_weights.is_empty() {
+            return Err(errstr("Empty Miniscript compilation"));
+        }
+        while node_weights.len() > 1 {
+            let (p1, s1) = node_weights.pop().expect("len must atleast be two");
+            let (p2, s2) = node_weights.pop().expect("len must atleast be two");
+
+            let p = (p1.0).0 + (p2.0).0;
+            node_weights.push((Reverse(OrdF64(p)), Arc::from(TapTree::Tree(s1, s2))));
+        }
+
+        debug_assert!(node_weights.len() == 1);
+        let node = node_weights
+            .pop()
+            .expect("huffman tree algorithm is broken")
+            .1;
+        Ok((*node).clone())
```

Comment:
`node` is of type `Arc<_>` while our return type is `TapTree<Pk>`.

![image](https://user-images.githubusercontent.com/53688395/149748621-2a5334c5-a028-440b-9dc7-7fbdf98edb3f.png)

As I understand, we can't move the value out of `Arc` and the only solution is to call `clone` on the reference for node, either directly in the function or at a later stage (since we require owned value of the node as `Option<TapTree<Pk>>` for `Descriptor::new_tr()` rather than a reference to it.)

![image](https://user-images.githubusercontent.com/53688395/149748931-76347e4b-48b7-4b75-8131-b08dacb9851d.png)

Reply:
suggested how to fix it above.